File: flutter_canli_yayin/lib/models/chat_message_model.dart
class ChatMessageModel {
  final String roomId;
  final String username;
  final String message;
  final String createdAt;

  ChatMessageModel({
    required this.roomId,
    required this.username,
    required this.message,
    required this.createdAt,
  });

  factory ChatMessageModel.fromJson(Map<String, dynamic> json) {
    return ChatMessageModel(
      roomId: json["room_id"] ?? "",
      username: json["username"] ?? "",
      message: json["message"] ?? "",
      createdAt: json["created_at"] ?? "",
    );
  }
}

File: CanliYayinSwiftUI/CanliYayinSwiftUI/model/ChatMessageModel.swift
//
//  ChatMessageModel.swift
//  CanliYayinSwiftUI
//
//  Created by Alperen Saraç on 14.05.2026.
//

import Foundation

struct ChatMessageModel: Identifiable {
    let id = UUID()
    let roomId: String
    let username: String
    let message: String
    let createdAt: String

    init(json: [String: Any]) {
        self.roomId = json["room_id"] as? String ?? ""
        self.username = json["username"] as? String ?? ""
        self.message = json["message"] as? String ?? ""
        self.createdAt = json["created_at"] as? String ?? ""
    }
}
